### PR TITLE
Degree Flow Refactorings

### PR DESCRIPTION
**Pull request by malcolmbaig**
## Context

Some small refactors identified during https://trello.com/c/zeuemNFz

## Changes proposed in this pull request

See each commit - they're self-contained so can be individually discarded if there's any disagreement/objections around these changes.

## Guidance to review
View the diff, per-commit recommended.

## Link to Trello card
na

## Things to check

- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: spec/system/candidate_interface/candidate_entering_degrees_spec.rb [hunk 1]
```diff
@@ -287,7 +287,10 @@ def when_i_click_to_change_my_undergraduate_degree_type
   end
 
   def when_i_click_to_change_my_undergraduate_degree_year
-    click_change_link('year')
+    start_year_row = find('.govuk-summary-list__row', text: 'Start year')
+    within start_year_row do
+      click_change_link('year')
+    end
```
Comment (stevehook) [inline spec/system/candidate_interface/candidate_entering_degrees_spec.rb:293]: :+1: Made me wonder, if this is a common pattern, could you go further with a helper that looks something like: `click_change_link('year', within: 'Start year')`? Maybe it's rare enough that it doesn't make sense.

**Review by stevehook — APPROVED** [review]
all really good stuff, no objections from me!

File: spec/system/candidate_interface/candidate_entering_degrees_spec.rb (hunk 1 above)
Comment (malcolmbaig) [reply spec/system/candidate_interface/candidate_entering_degrees_spec.rb:293]: Like that idea. Tested it by going through and using this helper across the candidate interface specs. I found the need to scope wasn't that common (and in a few cases, it was just as clear to leave the `within` call in the spec itself). 

Pushed those replacements in the newest commit. WDYT?

Comment (stevehook) [reply spec/system/candidate_interface/candidate_entering_degrees_spec.rb:293]: Looks good to me, I don't think it's worth adding the extra complexity to the helper for a rare case.

**Review by stevehook — APPROVED** [review]

**Review by tvararu — APPROVED** [review]

**Pull request merged**